packages/core/src/components/axes/toolbar.ts

```diff
@@ -0,0 +1,489 @@
+// Internal Imports
+import { Component } from "../component";
+import { Events, Roles, ToolbarControlTypes } from "../../interfaces";
+import { Tools } from "../../tools";
+import { DOMUtils } from "../../services";
+import * as Configuration from "../../configuration";
+
+// D3 Imports
+import { event, select } from "d3-selection";
+
+// import the settings for the css prefix
+import settings from "carbon-components/es/globals/js/settings";
+
+export class Toolbar extends Component {
+	type = "toolbar";
+
+	// overflow menu button to control background color
+	overflowButton: any;
+
+	// ul options list element
+	overflowMenu: any;
+
+	// x, y coordinate of overflow menu
+	overflowMenuX = 0;
+	overflowMenuY = 0;
+
+	// Use a random number to create overflow menu item unique ID
+	// so they don't interfere the other overflow menu item in a page
+	overflowMenuItemId = Math.floor(Math.random() * 99999999999);
+
+	init() {
+		const options = this.model.getOptions();
+
+		// Grab the tooltip element
+		const holder = select(this.services.domUtils.getHolder());
+		const chartPrefix = Tools.getProperty(options, "style", "prefix");
+
+		this.overflowMenu = DOMUtils.appendOrSelect(
+			holder,
+			`div.${settings.prefix}--${chartPrefix}--overflowMenu`
+		);
+
+		this.overflowMenu.style("max-width", null);
+
+		// listen to show overflow menu event to render the overflow menu
+		this.services.events.addEventListener(
+			Events.Toolbar.SHOW_OVERFLOW_MENU,
+			() => {
+				this.overflowMenu.html(this.getOverflowMenuHTML());
+			}
+		);
+
+		// listen to hide overflow menu event to hide the overflow menu
+		this.services.events.addEventListener(
+			Events.Toolbar.HIDE_OVERFLOW_MENU,
+			() => {
+				this.overflowMenu.html(null);
+			}
+		);
+		// hide overflow menu if user clicks on somewhere in web page
+		document.body.addEventListener("click", () =>
+			this.showOverflowMenu(false)
+		);
+	}
+
+	render(animate = true) {
+		const isDataLoading = Tools.getProperty(
+			this.model.getOptions(),
+			"data",
+			"loading"
+		);
+
+		// size of toolbar button with background
+		const buttonSize = Configuration.toolbar.buttonSize;
+		const parentY = parseFloat(this.parent.node().getAttribute("y"));
+
+		const svg = this.getContainerSVG();
+
+		// TODO -- adjust toolbar Y position to align title component
+		// before layout component supports vertical alignment center
+		const VERTICAL_ALIGNMENT_Y = -6;
```

Comment:
```suggestion
		const Y_OFFSET = -6;
```

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/791/commits/ffb41bcd7af6b329ba8ba5538c427491ad32453a.